cohttp/src/request.ml

```diff
@@ -27,9 +27,9 @@ type t = {
 let fixed_zero = Transfer.Fixed Int64.zero
 
 let guess_encoding ?(encoding=fixed_zero) headers =
-  match Header.get_content_range headers with
-  | Some clen -> Transfer.Fixed clen
-  | None -> encoding
+  match Header.get_transfer_encoding headers with
+  | Transfer.(Chunked | Fixed _) as enc -> enc
+  | Unknown -> encoding
```

Comment:
This looks good. Would it be possible to get a test case added to exercise this part of the RFC? I.e. one with the transfer encoding headers, and another with content range, and another with all three.  There are already some similar tests in the tests/ directory.